Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2227,6 +2227,41 @@ extension BrowserViewController: WKUIDelegate {
             self.tabManager.removeTab(tab)
         }
     }
+    
+    func webView(_ webView: WKWebView, shouldPreviewElement elementInfo: WKPreviewElementInfo) -> Bool {
+        guard let url = elementInfo.linkURL, !Storage.isIgnoredURL(url) else { return false }
+        return true
```

Comment:
wdyt about:

```
guard let url = elementInfo.linkURL else { return false }
return !Storage.isIgnoredURL(url)
```

Mostly because `guard`ing followed by a `!` which is negating a `isIgnored` is kind of a lot of logic shifts to follow.

Reply:
I also don't think we should use `Storage` framework in our code. It's a leftover from Firefoxes backend that we must remove at some point

Reply:
@jhreis absolutely 👍 I was even thinking about adding a method to `Storage` like `canProcessURL`, `isRecognizedURL` or `isSupportedURL` to avoid a negation `!` 🤔 

@iccub is there any other way that I can check if a given url is not ignored and we can preview it? 🤔 

Is there another component available in the code base I should have used instead of `Storage`?  🤔

🙇 🙇

Reply:
This is code for `isIgnoredURL`

```
private var ignoredSchemes = ["about"]

public func isIgnoredURL(_ url: URL) -> Bool {
    guard let scheme = url.scheme else { return false }

    if let _ = ignoredSchemes.index(of: scheme) {
        return true
    }

    if url.host == "localhost" {
        return true
    }

    return false
}
```

I would just move it outside of Storage framework(we want to completely remove Storage framework in the future)

Move it to `Shared` module in `URLExtensions`
and I would rename it to `eligibleForPeekAndPop`
so like

```
extension URL {
  var eligibleForPeekAndPop: Bool {
    let ignoredSchemes = ["about"]

    // rest of the method, replace 'url' with self since this is URL's extension
  }
}
```

Reply:
Sounds good 👍 

> // rest of the method, replace 'url' with self since this is URL's extension

We will also need to reverse the returned value:

```Swift
var eligibleForPeekAndPop: Bool {
        let ignoredSchemes = ["about"]
        
        guard let scheme = self.scheme else { return false }
        
        if let _ = ignoredSchemes.index(of: scheme) {
            return false
        }
        
        if self.host == "localhost" {
            return false
        }
        
        return true
    }
```

I will also add the unit tests for this extension 💪